snapcraft_legacy/_store.py

```diff
@@ -807,8 +825,9 @@ def _upload_delta(
             raise storeapi.errors.StoreDeltaApplicationError(str(e))
         else:
             raise
-    except storeapi.http_clients.errors.StoreServerError as e:
-        raise storeapi.errors.StoreUploadError(snap_name, e.response)
+    except craft_store.errors.StoreServerError as store_error:
+        if store_error.response.status_code == 401:
```

Comment:
this is wrong

Reply:
Store returning wrong status or wrong logic testing status?

Reply:
it is the wrong logic :-)